ts/src/profilers/profile-serializer.ts

```diff
@@ -140,7 +140,7 @@ function serialize<T extends ProfileNode>(
     }
     id = functions.length + 1;
     functionIdMap.set(keyStr, id);
-    let nameId = stringTable.getIndexOrAdd(node.name);
+    let nameId = stringTable.getIndexOrAdd(node.name || '(anonymous)');
```

Comment:
The only thing that slightly worries me is whether empty string always means anonymous function. Are there conditions when the runtime actually failed to provide the function name for some other reason? Should we rename the function to (anonymous) only if the script name is present, or that's overkill?

Reply:
/cc @hashseed: this is probably related to function name inference done by V8. What are the cases when we get an empty function name vs. `(anonymous)`?

Reply:
I don't think the function name inferrer ever returns "(anonymous)". It would be empty.

Reply:
@hashseed Is there a way to tell "good empty" (i.e. anonymous function) from "bad empty" (i.e. something failed while the function name was being inferred)?

Reply:
Looking at the V8 source code, I don't see the string `(anonymous)` anywhere. I am guessing that V8 returns empty string in anonymous case.

Looking at the DevTools source, it seems they use the [same logic](https://cs.chromium.org/chromium/src/out/Debug/gen/devtools/inspector.js?type=cs&q=%5C(anonymous%5C)&l=2997) as being proposed in this PR: infer `(anonymous)` when the name is empty.